developer-workflow/c-api.rst

```diff
@@ -110,9 +110,11 @@ Guidelines for expanding/changing the public API
   fields, arguments and return values are well defined.
 
 - Functions returning ``PyObject *`` must return a valid pointer on success,
-  and ``NULL`` with an exception raised on error.
-  Most other API must return ``-1`` with an exception raised on error,
-  and ``0`` on success.
+  and ``NULL`` if and only if an exception is raised.
+  Other API must return ``0`` on success,
+  and a negative value if and only if an exception is raised.
+  Unless there is need to distinguish between error types,
+  ``-1`` should be used as the error value.
```

Comment:
@encukou, I suggest either:

```suggestion
  and a ``-1`` if and only if an exception is raised.
```

... or:

```suggestion
  and a negative value if and only if an exception is raised.
```

Reply:
```suggestion
  and a negative value if and only if an exception is raised.
  ``-1`` should be used as the error value;
  the caller should use the C idiom ``if (func() < 0)`` to check for errors.
```

Reply:
I prefer to be specific about -1 and suggest testing func() < 0.